frontend/src/helpers/AppRoutes.jsx

```diff
@@ -12,107 +8,116 @@ import Polls from '../pages/polls/Polls';
+      <Route exact path='/create'>
+        {!isLoggedIn ? (
+          <Redirect to='/' />
+        ) : mustChangePassword ? (
+          <Redirect to='/changepassword' />
+        ) : isRegularUser ? (
+          <Create />
+        ) : (
+          <Redirect to='/' />
+        )}
+      </Route>
+      <Route exact path='/notifications'>
+        {!isLoggedIn ? (
+          <Redirect to='/' />
+        ) : mustChangePassword ? (
+          <Redirect to='/changepassword' />
+        ) : isRegularUser ? (
+          <Notifications />
+        ) : (
+          <Redirect to='/' />
+        )}
+      </Route>
+      <Route exact path='/profile'>
+        {isRegularUser ? <Profile /> : <Redirect to='/' />}
+      </Route>
 
-            {/* Private Routes for Regular Users Only */}
-            <Route path='/changepassword'>
-                {isRegularUser ? <ChangePassword /> : <Home />}
-            </Route>
-            <Route path='/create'>
-                {!isLoggedIn ? (
-                        <Home />
-                    ) : mustChangePassword ? (
-                        <Redirect to='/changepassword' />
-                    ) : isRegularUser ? (
-                        <Create />
-                    ) : (
-                        <Home /> // catch all for all other condtions, not a regular user
-                    )}
-            </Route>
-            <Route path='/notifications'>
-                {!isLoggedIn ? (
-                    <Home />
-                ) : mustChangePassword ? (
-                    <Redirect to='/changepassword' />
-                ) : isRegularUser ? (
-                    <Notifications />
-                ) : (
-                    <Home /> // catch all for all other condtions, not a regular user
-                )}
-            </Route>
-            <Route path='/profile'>
-                {isRegularUser ? <Profile /> : <Home />}
-            </Route>
-            {/* Admin Routes */}
-            <Route path='/admin'>
-                {isLoggedIn && auth.usertype === 'ADMIN' ? <AdminDashboard /> : <Home />}
-            </Route>
+      {/* Admin Routes */}
+      <Route exact path='/admin'>
+        {isLoggedIn && auth.usertype === 'ADMIN' ? (
+          <AdminDashboard />
+        ) : (
+          <Redirect to='/' />
+        )}
+      </Route>
 
-            {/* If no other route matches, redirect to home */}
-            <Route path='/'>
-            {isLoggedIn && !auth.usertype === 'ADMIN' && mustChangePassword ? (
-                    <Redirect to='/changepassword' />
-                ) : (
-                    <Home />
-                )}
-            </Route>
+      {/* Home Route */}
+      <Route exact path='/'>
+        {isLoggedIn && auth.usertype !== 'ADMIN' && mustChangePassword ? (
+          <Redirect to='/changepassword' />
+        ) : (
+          <Home />
+        )}
+      </Route>
 
-        </Switch>
-    );
+      {/* 404 Route - This should be the last route */}
```

Comment:
I added some comments on the overall, "discussion" part of this pull request.

https://github.com/openpredictionmarkets/socialpredict/pull/235#issuecomment-2266792516

Basically it seems that 404 is a last result, if a main route does not work. So this seems essential and good that we have this now. However, a couple things:

1. Will it confuse users if they get redirected to `/login` after first signing out, which seems to be the default now, and then they see a 404? See my images on the main discussion.
2.  @astrosnat had originally asked if it were possible to put a 404 on pages such as, `markets/99` for a market that has not been created yet. I don't know how complicated this would be to put in place. This could be a different, separate feature request if just simply getting the 404 request is the main priority ... e.g. probably better to do a small change at a time and have more tiny features put in place over time.